src/Microsoft.Graph.Core/Exceptions/ServiceException.cs

```diff
@@ -25,15 +35,23 @@ public ServiceException(Error error, System.Net.Http.Headers.HttpResponseHeaders
             this.ResponseHeaders = responseHeaders;
             this.StatusCode = statusCode;
         }
-        
+
         /// <summary>
         /// Creates a new service exception.
         /// </summary>
         /// <param name="error">The error that triggered the exception.</param>
         /// <param name="innerException">The possible innerException.</param>
-        public ServiceException(Error error, Exception innerException = null)
+        /// <param name="responseHeaders">The HTTP response headers from the response.</param>
+        /// <param name="statusCode">The HTTP status code from the response.</param>
+        /// <param name="rawResponseBody">The raw JSON response body.</param>
+        public ServiceException(Error error, 
```

Comment:
It looks like calling this constructor will cause some loss of information. 
The `statusCode` and `responseHeaders` parameters are reset to `default` and `null` respectively and will therefore not be passed to the new object being created.

Reply:
Agreed, this seems wrong.

Reply:
addressed